ckanext/validation/logic.py

```diff
@@ -217,12 +217,16 @@ def resource_validation_delete(context, data_dict):
         context, {u'id': data_dict[u'resource_id']})
 
     try:
-        resource.pop(u'validation_status')
-        resource.pop(u'validation_options')
-        resource.pop(u'validation_timestamp')
+        if resource.get('validation_status', False):
+            resource.pop(u'validation_status')
+        if resource.get('validation_options', False):
+            resource.pop(u'validation_options')
+        if resource.get('validation_timestamp', False):
+            resource.pop(u'validation_timestamp')
```

Comment:
Is there a reason you want to keep only Falsey-valued validation fields here?

Reply:
Hmmm no. Wanted to say `if in resource` as the pop was throwing key errors. Which was weird to me, but I do not use pop so not sure if it checks if the key exists in the dict or not before hand.

Will fix.

Reply:
Okay this should be good now. Things have been moved outside the try/catch and the action now uses `ckanapi.localCKAN`